File: packages/expo-updates/ios/EXUpdates/AppLoader/EXUpdatesManifestHeaders.m
//  Copyright © 2019 650 Industries. All rights reserved.

#import <EXUpdates/EXUpdatesManifestHeaders.h>

@implementation EXUpdatesManifestHeaders

- (instancetype)initWithProtocolVersion:(nullable NSString *)protocolVersion
                   serverDefinedHeaders:(nullable NSString *)serverDefinedHeaders
                        manifestFilters:(nullable NSString *)manifestFilters
                      manifestSignature:(nullable NSString *)manifestSignature
                              signature:(nullable NSString *)signature {
  if (self = [super init]) {
    _protocolVersion = protocolVersion;
    _serverDefinedHeaders = serverDefinedHeaders;
    _manifestFilters = manifestFilters;
    _manifestSignature = manifestSignature;
    _signature = signature;
  }
  return self;
}

@end

File: packages/expo-updates/ios/EXUpdates/AppLoader/EXUpdatesManifestHeaders.swift
//  Copyright © 2019 650 Industries. All rights reserved.

import Foundation

public final class EXUpdatesManifestHeaders {
  public let protocolVersion: String?
  public let serverDefinedHeaders: String?
  public let manifestFilters: String?

  /// Classic updates Expo Go manifest signature
  public let manifestSignature: String?

  /// Code signing manifest signature
  public let signature: String?

  public required init(protocolVersion: String?, serverDefinedHeaders: String?, manifestFilters: String?, manifestSignature: String?, signature: String?) {
    self.protocolVersion = protocolVersion
    self.serverDefinedHeaders = serverDefinedHeaders
    self.manifestFilters = manifestFilters
    self.manifestSignature = manifestSignature
    self.signature = signature
  }
}
